### DISCUSSION TITLE
Track Proofreading: Instances Inacessable in "Next Track Spawn Frame"

### DISCUSSION THREAD

**Discussion by mollywestbrook** [talmolab/sleap#1292] [Help!] [answered]
Hi everyone! Doing some track proofreading on a video, and I noticed that there's a track instance that was generated, but that I can't access through the 'next track spawn frame'. I can't click on the track itself within the GUI, but it lines up with a large gap in an existing track identity (screengrab below, yellow track_2, and the blue track below is my missing tracklet). When I exported the .h5 (terminal output below here), there were 6 track identities generated, but I dug into the file and discovered that track 6 was exclusively NaNs, except for a single frame of data. Any guidance for how to get my missing track data or general fixes to problems like this would be great. Let me know if there's any more info needed to solve. Thanks!

`ignoring 334 empty tracks
track_names: 6
node_names: 8
edge_names: 7
edge_inds: 7
tracks: (7485, 8, 2, 6)
track_occupancy: (6, 7485)
point_scores: (7485, 8, 6)
instance_scores: (7485, 6)
tracking_scores: (7485, 6)
labels_path: C:/Users/Junttilab/Desktop/SLEAP Pilot/labels.v002_20230324.slp
video_path: C:/Users/Junttilab/Desktop/SLEAP Pilot/videos/20230209_test2_5larva_15dpf_25fps.avi
video_ind: 0
provenance: {}
Saved as C:/Users/Junttilab/Desktop/SLEAP Pilot/labels.v002_20230324.000_20230209_test2_5larva_15dpf_25fps_complete.analysis.h5`

The screengrabs, here with track_2:
![sleap_screengrab](https://user-images.githubusercontent.com/48096260/235480851-dd4c3103-2dde-4376-bd96-191284cea8e4.png)

And here with missing track:
![image](https://user-images.githubusercontent.com/48096260/235481089-55f054fb-6369-437b-94c7-fd2eeebe3d3d.png)

**Comment by gitttt-1234** [comment, marked as answer]
Closing this discussion, solved in [`sleap-nn`](https://nn.sleap.ai/latest/guides/tracking/) tracking pipeline which has options to prevent ghost tracks:

```bash
sleap-track input.slp --model /path/to/model \
    --max_tracks 5 \
    --post_connect_single_breaks \
    --tracking_target_instance_count 5
```

- `--max_tracks 5` caps the number of tracks so extra detections get assigned to existing tracks
- `--post_connect_single_breaks` reconnects single-frame track gaps (exactly the yellow track_2 gap scenario)
